sdv/single_table/base.py

```diff
@@ -97,8 +100,15 @@ def _check_metadata_updated(self):
     def __init__(
         self, metadata, enforce_min_max_values=True, enforce_rounding=True, locales=['en_US']
     ):
+        single_metadata = metadata
+        if isinstance(single_metadata, Metadata):
+            single_metadata = single_metadata._convert_to_single_table()
+            self.real_metadata = metadata
+        elif isinstance(single_metadata, SingleTableMetadata):
```

Comment:
Don't forget to raise a `warning` As per the issue:

> * Deprecate SingleTableMetadata
    * Raise a future warning if users pass old metadata